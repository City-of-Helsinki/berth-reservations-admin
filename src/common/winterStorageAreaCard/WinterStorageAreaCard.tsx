import React from 'react';
import { useTranslation } from 'react-i18next';
import classNames from 'classnames';

import Card from '../card/Card';
import CardHeader from '../cardHeader/CardHeader';
import Text from '../text/Text';
import CardBody from '../cardBody/CardBody';
import placeholder from '../placeholderImage.svg';
import Section from '../section/Section';
import ExternalLink from '../externalLink/ExternalLink';
import MapLinks from '../mapLinks/MapLinks';
import Grid from '../grid/Grid';
import Property from '../property/Property';
import { IconDollyEmpty, IconFence, IconPlug, IconTrestle, IconWaterTap, IconBoat } from '../icons';
import { WinterStorageArea } from '../../features/winterStorageAreaView/types';
import { getProductServiceTKey } from '../utils/translations';
import { ProductServiceType } from '../../@types/__generated__/globalTypes';
import styles from './winterStorageAreaCard.module.scss';

interface WinterStorageAreaCardProps extends WinterStorageArea {
  className?: string;
}

const WinterStorageAreaCard = ({
  className,
  name,
  servicemapId,
  imageFile,
  streetAddress,
  zipCode,
  municipality,
  numberOfCustomers,
  gate,
  water,
  electricity,
  summerStorageForBoats,
  summerStorageForDockingEquipment,
  summerStorageForTrailers,
  maps,
}: WinterStorageAreaCardProps) => {
  const { t } = useTranslation();
  const serviceMapUrl = `${process.env.REACT_APP_SERVICE_MAP_URI}${servicemapId}`;

  return (
    <Card className={classNames(className, styles.card)}>
      <CardHeader title={t('winterStorageAreaCard.title')} />
      <CardBody>
        <div className={styles.cardBody}>
          <div className={styles.details}>
            <div className={styles.imageWrapper}>
              <img
                alt={t('winterStorageAreaCard.imgAlt')}
                src={imageFile ? imageFile : placeholder}
                className={styles.image}
              />
            </div>
            <div>
              <Section>
                <Text as="h2" size="xxl">
                  {name}
                </Text>
                <Text>
                  {streetAddress ? `${streetAddress}, ${zipCode} ${municipality}` : `${zipCode} ${municipality}`}
                </Text>
              </Section>
              <Section>
                <ExternalLink href={serviceMapUrl} variant="withArrow">
                  {t('common.terminology.serviceMap')}
                </ExternalLink>
              </Section>
              <MapLinks maps={maps} />
            </div>
          </div>
          <Grid colsCount={6} className={styles.propsGrid}>
            <div />
            <div />
            <div />
            <div />
            <div />
            <Property counter={numberOfCustomers} label={t('winterStorageAreaCard.numberOfCustomers')} />

            <Property
              icon={IconPlug}
              label={t(getProductServiceTKey(ProductServiceType.ELECTRICITY))}
              active={electricity}
              className={styles.wordBreak}
            />
            <Property
              icon={IconFence}
              label={t(getProductServiceTKey(ProductServiceType.GATE))}
              active={gate}
              className={styles.wordBreak}
            />
            <Property
              icon={IconWaterTap}
              label={t(getProductServiceTKey(ProductServiceType.WATER))}
              active={water}
              className={styles.wordBreak}
            />
            <Property
              icon={IconBoat}
              label={t('common.terminology.boatSummerStorage')}
              active={summerStorageForBoats}
              className={styles.wordBreak}
            />
            <Property
              icon={IconDollyEmpty}
              label={t(getProductServiceTKey(ProductServiceType.SUMMER_STORAGE_FOR_TRAILERS))}
              active={summerStorageForTrailers}
              className={styles.wordBreak}
            />
            <Property
              icon={IconTrestle}
              label={t(getProductServiceTKey(ProductServiceType.SUMMER_STORAGE_FOR_DOCKING_EQUIPMENT))}
              active={summerStorageForDockingEquipment}
              className={styles.wordBreak}
            />
          </Grid>
        </div>
      </CardBody>
    </Card>
  );
};

export default WinterStorageAreaCard;
