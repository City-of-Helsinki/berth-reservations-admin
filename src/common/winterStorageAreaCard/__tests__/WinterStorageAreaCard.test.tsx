import React from 'react';
import { shallow } from 'enzyme';

import WinterStorageAreaCard from '../WinterStorageAreaCard';

describe('WinterStorageAreaCard', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <WinterStorageAreaCard
        name={'name'}
        gate={true}
        electricity={true}
        servicemapId={'12345'}
        streetAddress={'street'}
        zipCode={'00000'}
        municipality={'municipality'}
        imageFile={''}
        maps={[]}
        numberOfCustomers={0}
        summerStorageForBoats={false}
        summerStorageForDockingEquipment={true}
        summerStorageForTrailers={false}
        water={true}
        wwwUrl={'www.google.com'}
      />
    );
    expect(wrapper.render()).toMatchSnapshot();
  });
});
