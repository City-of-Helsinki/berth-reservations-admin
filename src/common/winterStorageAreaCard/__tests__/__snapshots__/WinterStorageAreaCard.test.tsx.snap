// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WinterStorageAreaCard renders correctly 1`] = `
<div
  class="card card"
>
  <div
    class="cardHeader"
  >
    <span
      class="title"
    >
      TALVISÄILYTYSALUE
    </span>
  </div>
  <div
    class="cardBody"
  >
    <div
      class="cardBody"
    >
      <div
        class="details"
      >
        <div
          class="imageWrapper"
        >
          <img
            alt="Talvisäilytyspaikan sijainti"
            class="image"
            src="placeholderImage.svg"
          />
        </div>
        <div>
          <article
            class="section"
          >
            <section
              class="body"
            >
              <h2
                class="text standard h2 xxl"
              >
                name
              </h2>
              <span
                class="text standard span"
              >
                street, 00000 municipality
              </span>
            </section>
          </article>
          <article
            class="section"
          >
            <section
              class="body"
            >
              <a
                class="link withArrow"
                href="https://palvelukartta.hel.fi/unit/12345"
                rel="noopener noreferrer"
                target="_blank"
              >
                Palvelukartta
                <svg
                  class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_m__3edUY icon_hds-icon--size-m__1mcHv arrow"
                  color="brand"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    fill="none"
                    fill-rule="evenodd"
                  >
                    <path
                      d="M0 24V0h24v24z"
                    />
                    <path
                      d="M13 12.5l-5-5L9.5 6l6.5 6.5L9.5 19 8 17.5z"
                      fill="currentColor"
                    />
                  </g>
                </svg>
              </a>
            </section>
          </article>
        </div>
      </div>
      <div
        class="grid cols6 propsGrid"
      >
        <div />
        <div />
        <div />
        <div />
        <div />
        <div
          class="property"
        >
          <strong
            class="text standard strong xxxl"
          >
            0
          </strong>
          <span
            class="text standard span s bold label"
          >
            Asiakasta
          </span>
        </div>
        <div
          class="property wordBreak"
        >
          <div
            class="iconWrapper standard outlined"
          >
            <svg
              class="icon m"
              height="100%"
              viewBox="0 0 400 400"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M159.86 380.84h42v74.41h-42zM327.33 142.77H262.1v-81h-32v81h-98.47v-81h-32v81H34.39v32H67v48.82a111.59 111.59 0 0 0 21.6 65.91 119.83 119.83 0 0 0 43.63 36.86v42.86h97.24v-42.5a113.69 113.69 0 0 0 65.23-103.13v-48.82h32.62zm-64.62 80.82c0 34.94-21.88 66-54.46 77.24l-10.77 3.72v32.67h-33.24V305L154 301c-32.91-12.8-55-43.92-55-77.45v-48.78h163.7z"
              />
            </svg>
          </div>
          <span
            class="text standard span s bold label"
          >
            Sähkö
          </span>
        </div>
        <div
          class="property wordBreak"
        >
          <div
            class="iconWrapper standard outlined"
          >
            <svg
              class="icon m"
              height="100%"
              viewBox="0 0 480 480"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M410.18 85l-62 46.23v60.5h-25v-60.44L262.1 85l-62 46.23v60.5h-25v-60.44L114 85l-62 46.22v332.17h123.16V383.5h25v79.89h123.09V383.5h25v79.89h123.08v-332.1zm-178.06 62.27L262 125l29.3 22.18v44.52h-59.18zm59.13 76.45V351.5h-59.13V223.72zM143.16 431.39H84V147.27L113.87 125l29.29 22.18v44.52h-41v32h41v127.8h-41v32h41zm32-79.89V223.72h25V351.5zm116.09 79.89h-59.13V383.5h59.13zm32-79.89V223.72h25V351.5zm116.08 79.89h-59.12V383.5h39v-32h-39V223.72h39v-32h-39v-44.45L410 125l29.29 22.18z"
              />
            </svg>
          </div>
          <span
            class="text standard span s bold label"
          >
            Portti
          </span>
        </div>
        <div
          class="property wordBreak"
        >
          <div
            class="iconWrapper standard outlined"
          >
            <svg
              class="icon m"
              height="100%"
              viewBox="0 0 480 480"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M154.61 313.9l-11.36-11.45-11.36 11.45a341.93 341.93 0 0 0-26.29 30.56c-18.95 25-28.16 45.16-28.16 61.73C77.44 425 84.87 442 98.37 454a68.35 68.35 0 0 0 89.76 0c13.5-12 20.94-29 20.94-47.81 0-16.57-9.21-36.76-28.16-61.73a341.09 341.09 0 0 0-26.3-30.56zm-11.36 124.9c-22.2 0-33.81-16.4-33.81-32.61 0-4.14 2.09-16.54 21.44-42.11 4.29-5.66 8.61-10.88 12.37-15.22 3.76 4.34 8.09 9.56 12.37 15.22 19.36 25.57 21.45 38 21.45 42.11 0 16.21-11.62 32.61-33.82 32.61zM341.91 158.33v-44h66.42v-32H243.49v32h66.42v44h-83.63c-57.36 0-104 45.76-104 102v21h42v-21c0-33.1 27.83-60 62-60h220.26v-42z"
              />
            </svg>
          </div>
          <span
            class="text standard span s bold label"
          >
            Vesi
          </span>
        </div>
        <div
          class="property wordBreak"
        >
          <div
            class="iconWrapper secondary outlined"
          >
            <svg
              class="icon m"
              height="100%"
              viewBox="0 0 24 24"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                id="prefix__prefix__Symbols"
              >
                <g
                  id="prefix__prefix__app_x2F_navigation-sidebar"
                  transform="translate(-21 -24)"
                >
                  <g
                    id="prefix__prefix__Venepaikat"
                    transform="translate(0 16)"
                  >
                    <g
                      id="prefix__prefix__atoms_x2F_icons_x2F_ship_x2F_24"
                      transform="translate(20 3.5)"
                    >
                      <g
                        id="prefix__prefix__Icon-color"
                        transform="translate(1 5)"
                      >
                        <g
                          id="prefix__prefix__Mask"
                        >
                          <path
                            d="M22 15.1l-1.5-4H1l2.9 4.4c-.1.1-.2.1-.3.2-.6.4-1 .7-1.8.7v1.5c1.2 0 2-.5 2.6-.9.6-.4 1-.7 1.8-.7.7 0 1.2.3 1.8.7.6.4 1.4.9 2.6.9 1.2 0 2-.5 2.6-.9.6-.4 1-.7 1.8-.7s1.2.3 1.8.7c.7.4 1.4.9 2.6.9 1.2 0 2-.5 2.6-.9.6-.4 1-.7 1.8-.7v-1.5c-.8-.1-1.4.1-1.8.3m-4.5.5c-.6-.4-1.4-.9-2.5-.9-1.2 0-1.9.5-2.5.9-.6.4-1 .7-1.8.7s-1.2-.3-1.8-.7c-.6-.4-1.4-.9-2.5-.9-.4 0-.7 0-1 .1l-1.5-2.3h15.8l1.2 3.3c-.4.3-.8.4-1.5.4-.9.1-1.4-.2-1.9-.6"
                          />
                          <path
                            d="M19.1 9V4.5h-3.2l-2.1 3.1h-7l-.9 1.5L19.1 9zm-2.5-3h1v1.7h-2.1L16.6 6z"
                          />
                        </g>
                      </g>
                    </g>
                  </g>
                </g>
              </g>
            </svg>
          </div>
          <span
            class="text secondary span s bold label"
          >
            Veneiden kesäsäilytys
          </span>
        </div>
        <div
          class="property wordBreak"
        >
          <div
            class="iconWrapper secondary outlined"
          >
            <svg
              class="icon m"
              height="100%"
              viewBox="0 0 480 480"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M456.93 233.27l-10.42-10.42.14-.13-6.47-6.2-8.7-8.7-.19.18-28-26.84-24.91 26 27.46 26.31-35.94 36a70.3 70.3 0 0 0-85.9-5.94l-97.31-97.31c-19.2-19.21-42.56-27.17-67.53-23-21 3.47-42.28 15.54-61.66 34.92l-12.74 12.69 25.45 25.46 12.73-12.73c16.84-16.83 49.39-40.77 78.27-11.89l99 99a70.19 70.19 0 0 0 112.56 80.69 70.22 70.22 0 0 0 17.16-71.08zM347.31 345.91a34.16 34.16 0 1 1 0-48.33 34.22 34.22 0 0 1 0 48.33z"
              />
            </svg>
          </div>
          <span
            class="text secondary span s bold label"
          >
            Trailerin kesäsäilytys
          </span>
        </div>
        <div
          class="property wordBreak"
        >
          <div
            class="iconWrapper standard outlined"
          >
            <svg
              class="icon m"
              height="100%"
              viewBox="0 0 480 480"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M386.18 428.65l33-14.47-99.78-227.27h56.29V83.35H134.31v103.56h58.29L140 306.83h-.27v.61L92.86 414.18l33 14.47 37.67-85.82h185zm-206.9-121.82l52.63-119.92h1.52v-3.46l.33-.75-.33-.14v-31.65h-63.12v-31.56h169.38v31.56h-60.92v31.56l-.53.23.53 1.2v3h1.32l52.63 119.92z"
              />
            </svg>
          </div>
          <span
            class="text standard span s bold label"
          >
            Telakointi­tarvikkeiden kesäsäilytys
          </span>
        </div>
      </div>
    </div>
  </div>
   
</div>
`;
